crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/service/CompareContributionServiceTest.java

```diff
@@ -56,6 +56,39 @@ void givenNoPreviousContribution_whenShouldCreateContributionIsInvoked_thenRetur
 
     @Test
     void givenActiveNotIdenticalContribution_whenShouldCreateContributionIsInvoked_thenReturnTrue() {
+        CalculateContributionDTO calculateContributionDTO =
+            TestModelDataBuilder.getContributionDTOForCompareContributionService(
+                CaseType.COMMITAL.getCaseTypeString(),
+                BigDecimal.valueOf(250),
+                BigDecimal.valueOf(250),
+                BigDecimal.valueOf(250),
+                LocalDate.now(),
+                MagCourtOutcome.APPEAL_TO_CC
+            );
+        ContributionResult contributionResult = ContributionResult.builder()
+            .totalAnnualDisposableIncome(BigDecimal.valueOf(16000.00))
+            .monthlyAmount(BigDecimal.valueOf(300.00))
```

Comment:
Can we make this a parameterised test? I notice that we're not covering all of the conditions in the `areContributionRecordsIdentical()` method, just the monthly amount, but the upfront amount, contribution cap and effective date are also part of the check.

Reply:
I'm going to go one better I think and move the equality check into the _ContributionResult_ record, otherwise it's not going to be truly comparing equality if some other check is introduced in `shouldCreateContributionIsInvoked` that changes the result (as we can't test `areContributionRecordsIdentical` because it's private).

Reply:
Update: parameterised as you suggested. I didn't implement the override on the `equals` method of _ContributionResult_ as I was originally going to as we're comparing different types in `areContributionRecordsIdentical`.